test/goaway_test.go

```diff
@@ -816,6 +818,7 @@ func (s) TestClientSendsAGoAway(t *testing.T) {
 			}
 		}
 	}()
+	cc.WaitForStateChange(ctx, connectivity.Connecting)
```

Comment:
what would be the max wait here? Timeout for all tests is 7m. Is there a chance it can breach that?

Reply:
Max timeout is as long context is not timed out.

Reply:
shouldn't the state to wait be `connectivity.Ready`?

Reply:
I think it depends where the race is happening.

If there's just a race where we call `Close` before we even start creating the connection, then this is a problem with the test, and `Connecting` should fix that.

If this is a race where a connection is established and doesn't get a GOAWAY written to it (due to internal state in the `ClientConn`/`addrConn`, then `Connecting` will expose that race and we should fix it.

Reply:
@purnesh42H WaitForStateChange waits until the connectivity.State of ClientConn changes from sourceState or ctx expires. A true value is returned in former case and false in latter. So we are providing sourceState as Connecting and hence it will wait until state changes.

Reply:
I think race is detected when client conn is getting closed before we try to send goAway so believe just waiting for clientConn state change will fix this race condition.